### PR TITLE
Tide: Fall back to v3 api for getting checkruns

### PR DESCRIPTION
**Pull request by alvaroaleman**
Fixes https://github.com/kubernetes/test-infra/issues/15611
Fixes https://github.com/kubernetes/test-infra/issues/19270
/assign @stevekuznetsov

### CODE REVIEW

**Comment by alvaroaleman** [comment]
/hold
the v4 change has been reverted because it increases token usage too much. That would need to be re-introduced first.

**Comment by alvaroaleman** [comment]
/hold cancel
the v4 change is back in

**Comment by alvaroaleman** [comment]
Why did this become so flaky :/
/retest

File: prow/ANNOUNCEMENTS.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## New features
 
 New features added to each component:
+  - *October 30, 2020* Tide is now able to respect checkruns
```
Comment (hi-rustin) [inline prow/ANNOUNCEMENTS.md:6]: Better to add a period.

**Review by stevekuznetsov — APPROVED** [review]
/hold

**Comment by BenTheElder** [comment]
So users just need to:
a) update tide
b) configure github actions (or other checks API statues) as "required" (e.g. with brancprotector?)
?

**Comment by alvaroaleman** [comment]
a) yes
b) The default is to respect all existing contexts tide doesn't know anything about, but also allowing to explicitly mark third-party contexts as required, required-if-present or optional: https://github.com/kubernetes/test-infra/blob/6300d86266ef9505c9d9cc5bf85efaf9d8b9419a/prow/config/prow-config-documented.yaml#L547

**Comment by BenTheElder** [comment]
thanks!

**Pull request merged**